src/bidiMapper/modules/network/NetworkProcessor.ts

```diff
@@ -467,6 +471,69 @@ export class NetworkProcessor {
     }
     return error;
   }
+
+  async addDataCollector(
+    params: Network.AddDataCollectorParameters,
+  ): Promise<Network.AddDataCollectorResult> {
+    if (params.userContexts !== undefined && params.contexts !== undefined) {
+      throw new InvalidArgumentException(
+        "'contexts' and 'userContexts' are mutually exclusive",
+      );
+    }
+    if (params.userContexts !== undefined) {
+      // Assert the user contexts exist.
+      await this.#userContextStorage.verifyUserContextIdList(
+        params.userContexts,
+      );
+    }
+    if (params.contexts !== undefined) {
+      for (const browsingContextId of params.contexts) {
+        // Assert the browsing context exists and are top-level.
+        const browsingContext =
+          this.#browsingContextStorage.getContext(browsingContextId);
+        if (!browsingContext.isTopLevelContext()) {
+          throw new InvalidArgumentException(
+            `Data collectors are available only on top-level browsing contexts`,
+          );
+        }
+      }
+    }
+    const collectorId = this.#networkStorage.addDataCollector(params);
+
+    // Enable fetch domain on impacted CDP targets.
+    await Promise.all(
```

Comment:
currently, the Network is unconditionally enabled, we can keep this code but inside toggleNetwork we do not need to turn on Fetch